massa-protocol-worker-2/src/handlers/block_handler/retrieval.rs

```diff
@@ -0,0 +1,1280 @@
+                                println!("Error: message not fully consumed");
+                                return;
+                            }
+                            match message {
+                                BlockMessage::AskForBlocks(block_infos) => {
+                                    if let Err(err) = self.on_asked_for_blocks_received(peer_id.clone(), block_infos) {
+                                        warn!("Error in on_asked_for_blocks_received: {:?}", err);
+                                    }
+                                }
+                                BlockMessage::ReplyForBlocks(block_infos) => {
+                                    for (block_id, block_info) in block_infos.into_iter() {
+                                        if let Err(err) = self.on_block_info_received(peer_id.clone(), block_id, block_info) {
+                                            warn!("Error in on_block_info_received: {:?}", err);
+                                        }
+                                    }
+                                    if let Err(err) = self.update_ask_block() {
+                                        warn!("Error in update_ask_blocks: {:?}", err);
+                                    }
+                                }
+                                BlockMessage::BlockHeader(header) => {
+                                    massa_trace!(BLOCK_HEADER, { "peer_id": peer_id, "header": header});
+                                    if let Ok(Some((block_id, is_new))) =
+                                        self.note_header_from_peer(&header, &peer_id)
+                                    {
+                                        if is_new {
+                                            self.consensus_controller
+                                                .register_block_header(block_id, header);
+                                        }
+                                        if let Err(err) = self.update_ask_block() {
+                                            warn!("Error in update_ask_blocks: {:?}", err);
+                                        }
+                                    } else {
+                                        warn!(
+                                            "peer {} sent us critically incorrect header, \
+                                            which may be an attack attempt by the remote peer \
+                                            or a loss of sync between us and the remote peer",
+                                            peer_id,
+                                        );
+                                        let _ = self.ban_node(&peer_id);
+                                    }
+                                }
+                            }
+                        },
+                        Err(err) => {
+                            println!("Error in retrieval block handler: {:?}", err);
```

Comment:
maybe error! ?

Reply:
This are not really errors because it's when the sender are dropped. Changed the print

Reply:
ok :)